src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,61 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isInterface()) {
+            throw new RuntimeException("Only interfaces are supported");
+        }
+
+        Class<?> implClass = findImplementation(interfaceClazz);
+        if (implClass == null || !implClass.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("No implementation found for " + interfaceClazz.getName());
```

Comment:
The exception message should be more informative and include the reason why the implementation wasn't found. According to the checklist, the message should indicate the missing `@Component` annotation if that's the case.